app/controllers/my_notifications.js

```diff
@@ -60,26 +93,42 @@ export default offers.extend({
 
   actions: {
     view(messageId) {
-      var message = this.store.peekRecord('message', messageId);
+      var message = this.store.peekRecord("message", messageId);
       var route = this.get("messagesUtil").getRoute(message);
       this.transitionToRoute.apply(this, route);
     },
 
     markThreadRead(notification) {
-      if(notification.unreadCount === 1) {
-        var message = this.store.peekRecord('message', notification.id);
+      if (notification.unreadCount === 1) {
+        var message = this.store.peekRecord("message", notification.id);
         this.get("messagesUtil").markRead(message);
       } else {
         this.send("view", notification.id);
       }
     },
 
     toggleShowUnread() {
-      this.set('showUnread', !this.get('showUnread'));
+      let showUnread = !this.get("showUnread");
+
+      if (!showUnread && !this.get("hasLoadedReadMessages")) {
```

Comment:
when we click on 'show all notifications' we query the read messages from the server